api/lib/source/polkadotV0-source.js

```diff
@@ -667,6 +678,224 @@ class polkadotAPI {
+        depositer: second,
+        amount: [
+          {
+            amount: toViewDenom(this.network, proposal.balance),
+            denom: this.network.stakingDenom
+          }
+        ]
+      }))
+    )
+    const votes = proposal.seconds.map((secondAddress) => ({
+      voter: secondAddress,
+      option: `Yes`
+    }))
+    const votesSum = proposal.seconds.length
+    return {
+      deposits,
+      depositsSum,
+      votes,
+      votesSum,
+      votingPercentageYes: `100`,
+      votingPercentagedNo: `0`,
+      links,
+      timeline: [{ title: `Proposal created`, time: proposal.creationTime }]
+    }
+  }
+
+  async getReferendumThreshold(proposal) {
+    const api = await this.getAPI()
+    let votingThresholdYes
+
+    const electorate = await api.query.balances.totalIssuance()
+    const ayeVotesWithoutConviction = proposal.allAye.reduce(
+      (ayeAggregator, aye) => {
+        return (ayeAggregator += Number(aye.balance))
+      },
+      0
+    )
+    const nayVotesWithoutConviction = proposal.allNay.reduce(
+      (nayAggregator, nay) => {
+        return (nayAggregator += Number(nay.balance))
+      },
+      0
+    )
+    const ayesVotes = Number(proposal.status.tally.ayes)
+    const naysVotes = Number(proposal.status.tally.nays)
+    if (
+      JSON.stringify(proposal.status.threshold) ===
+      JSON.stringify(`Supermajorityapproval`)
+    ) {
+      votingThresholdYes = getPassingThreshold({
+        nays: new BN(
+          BigNumber(naysVotes)
+            .times(this.network.coinLookup[0].chainToViewConversionFactor)
+            .toNumber()
+        ),
+        naysWithoutConviction: new BN(
+          BigNumber(nayVotesWithoutConviction)
+            .times(this.network.coinLookup[0].chainToViewConversionFactor)
+            .toNumber()
+        ),
+        totalIssuance: new BN(
+          BigNumber(electorate)
+            .times(this.network.coinLookup[0].chainToViewConversionFactor)
+            .toNumber()
+        ),
+        threshold: `Supermajorityapproval`
+      }).passingThreshold
+    }
+    if (
+      JSON.stringify(proposal.status.threshold) ===
+      JSON.stringify(`Supermajorityagainst`)
+    ) {
+      votingThresholdYes = getFailingThreshold({
+        ayes: new BN(
+          BigNumber(ayesVotes)
+            .times(this.network.coinLookup[0].chainToViewConversionFactor)
+            .toNumber()
+        ),
+        ayesWithoutConviction: new BN(
+          BigNumber(ayesVotesWithoutConviction)
+            .times(this.network.coinLookup[0].chainToViewConversionFactor)
+            .toNumber()
+        ),
+        totalIssuance: new BN(
+          BigNumber(electorate)
+            .times(this.network.coinLookup[0].chainToViewConversionFactor)
+            .toNumber()
+        ),
+        threshold: `Supermajorityagainst`
+      }).passingThreshold
+    }
+    if (
+      JSON.stringify(proposal.status.threshold) ===
+      JSON.stringify(`Simplemajority`)
```

Comment:
stringify a string?

Reply:
It is the only way to be 100% when you compare objects. In this case, ok, `Simplemajority` ist just a string, but curiously `proposal.status.threshold === 'Simplemajority'` won't work. Because of how the data is returned from polkadot/api:

I tried quite a few things and this one is the only one that worked :shrug: yes, another shrug here xD